File: src/main/java/com/study/platform/global/auth/adapter/KakaoOAuthClientAdapter.java
package com.study.platform.global.auth.adapter;

import com.study.platform.domain.user.dto.response.OAuthUserInfo;
import com.study.platform.global.auth.port.OAuthClient;
import com.study.platform.global.oauth.kakao.KakaoOAuthClient;
import com.study.platform.global.oauth.kakao.dto.KakaoIdTokenPayload;
import com.study.platform.global.oauth.kakao.dto.KakaoTokenResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class KakaoOAuthClientAdapter implements OAuthClient {

    private final KakaoOAuthClient kakaoOAuthClient;

    @Override
    public OAuthUserInfo getUserInfo(String code) {
        KakaoTokenResponse token = kakaoOAuthClient.getToken(code);
        KakaoIdTokenPayload payload = kakaoOAuthClient.parseIdToken(token.idToken());
        return new OAuthUserInfo(payload.sub(), payload.nickname(), payload.email());
    }
}

File: infrastructure/src/main/kotlin/com/study/platform/global/auth/adapter/KakaoOAuthClientAdapter.kt
package com.study.platform.global.auth.adapter

import com.study.platform.domain.user.dto.response.OAuthUserInfo
import com.study.platform.global.auth.port.OAuthClient
import com.study.platform.global.oauth.kakao.KakaoOAuthClient
import org.springframework.stereotype.Component

@Component
class KakaoOAuthClientAdapter(
    private val kakaoOAuthClient: KakaoOAuthClient
) : OAuthClient {

    override fun getUserInfo(code: String): OAuthUserInfo {
        val token = kakaoOAuthClient.getToken(code)
        val payload = kakaoOAuthClient.parseIdToken(token.idToken)
        return OAuthUserInfo(payload.sub, payload.nickname, payload.email)
    }
}
